app/lib/actions/pulp/repository/presenters/puppet_presenter.rb

```diff
@@ -42,7 +45,8 @@ def task_progress
           end
 
           def task_progress_details
-            task_progress && task_progress['puppet_importer']
+            return task_progress unless task_progress.key?('puppet_importer')
+            task_progress['puppet_importer']
```

Comment:
This would probably do, wouldn't it? `task_progress['puppet_importer'] || task_progress`